### DISCUSSION TITLE
Configure with RENOVATE_CONFIG environment variable is not seen by requireConfig

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#9972] [Suggest an Idea]
### Discussed in https://github.com/renovatebot/renovate/discussions/9970

I have configured renovate to run for my project in the internal Gitlab installation we have.

The project imports a centrally managed config that contains among other things:
```
variables:
    RENOVATE_EXTRA_FLAGS: ${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}
    RENOVATE_REQUIRE_CONFIG: 'true'
    RENOVATE_ONBOARDING: 'false'
    RENOVATE_CONFIG: '{ "extends": [ "local>developertools/renovate-config" ] }'
```
Now in the project to be checked there is no config file but it does have the config provided by the environment variable.

Since the documentation I found about the requireConfig says this I expect the project to be valid:
```
    name: 'requireConfig',
    description: 'Set to true if repositories must have a config to activate.',
```
Note the `self-hosted-configuration.md` does not have any information about this variable.

Yet when I run this config is then successfully read by the system (I see the company specific parameters set in the  "local>developertools/renovate-config" ) but later on it stops with 
```
DEBUG: config file not found (repository=developertools/renovate-experiment)
 INFO: Repository is disabled - skipping (repository=developertools/renovate-experiment)
```

I tracked this down to this code fragment

```
export const isOnboarded = async (config: RenovateConfig): Promise<boolean> => {
  logger.debug('isOnboarded()');
  const title = `Action required: Add a Renovate config`;
  // Repo is onboarded if admin is bypassing onboarding and does not require a
  // configuration file.
  if (config.requireConfig === false && config.onboarding === false) {
    // Return early and avoid checking for config files
    return true;
  }
  if (await configFileExists()) {
    await platform.ensureIssueClosing(title);
    return true;
  }
  logger.debug('config file not found');
```

Here the check ONLY ensures the config is in a **file**. The environment variable is disregarded.

My thoughts about this are that having the config in the environment variable should be valid too in case the onboarding is disabled. I'm not sure about the case when the onboarding is enabled (perhaps then it must be a file, not sure about that one).

**Comment by HonkingGoose** [comment]
Relevant code example:

https://github.com/renovatebot/renovate/blob/a70aa6b75bc812b2b21977e5565f245e53630768/lib/workers/repository/onboarding/branch/check.ts#L51-L69

**Comment by rarkins** [comment]
Moving this to a discussion as it is behaving per design.

**Comment by viceice** [comment]
This works as intended, by default renovate will require a config in repository before it will do anything, if it is missing, it will open an onboarding pr, which can be suppressed by `onboarding=false`. So then you need to manual onboard your repo.

If you like to fully configure renovate from bot config, you also need to set `requireConfig=false`. So i would suggest to update the `requireConfig` description

**Reply by rarkins** [reply]
If you set this combination:

```
    RENOVATE_REQUIRE_CONFIG: 'true'
    RENOVATE_ONBOARDING: 'false'
```

Then it means Renovate will skip a repository unless the repository has manually added a `renovate.json` or similar file. Some users have a valid reason for this and it should remain as-is.

In your case, if you want Renovate to run regardless of whether is a config file in the repository, why change `RENOVATE_REQUIRE_CONFIG` at all? With default values:

- Renovate will run on any repo it's been added to
- Renovate will not create an onboarding PR
- If a config file is present, it will parse it and any settings there add to or override your bot settings
- If no config file is present then the bot settings will be applied solely
